components/automate-ui/src/app/components/sidebar/sidebar.component.ts

```diff
@@ -15,12 +15,36 @@ export class SidebarComponent {
   constructor(
     @Inject(LayoutFacadeService) public layoutFacade: LayoutFacadeService
   ) {
-    this.menuGroups$ = layoutFacade.menuGroups$;
+    this.menuGroups$ = layoutFacade.sidebar$;
+    this.updateMenuGroupVisibility();
   }
 
-  public isVisible(item: any) {
-    return item.subscribe
-      ? item
-      : new BehaviorSubject(item);
+  public isAuthorized($event, menuItem, menuGroup) {
+    menuItem.authorized.isAuthorized = $event;
+    this.setGroupVisibility(menuGroup);
+  }
+
+  public hasMenuItemsNotRequiringAuthorization(menuItemGroup: any): boolean {
+    return menuItemGroup.items.some(menuItem =>
+      menuItem.authorized === undefined);
+  }
+
+  private updateMenuGroupVisibility(): void {
+    this.menuGroups$.subscribe((menuGroups: MenuItemGroup[]) => {
+      if (menuGroups) {
+        menuGroups.forEach(menuItemGroup => this.setGroupVisibility(menuItemGroup));
+      }
+    });
+  }
+
+  private setGroupVisibility(menuItemGroup: MenuItemGroup): void {
+    menuItemGroup.hasVisibleMenuItems =
+      this.hasVisibleMenuItems(menuItemGroup) ||
```

Comment:
Don't mean to belabor the point. 😁  This line probably should not have been changed: it is clearer if this reads 
```suggestion
      this.hasAuthorizedMenuItems(menuItemGroup) ||
```
Thus, we are then saying the `hasVisibleMenuItems` property (L41) is determined by whether there are authorized items (L42) or there are items that always show (L43).

Reply:
@msorens updated change :)